url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/expert_system/Rhythmic_Block.java

```diff
@@ -0,0 +1,805 @@
+	// #					   	METHODS			   				#
+	// ##########################################################
+	// Returns beatValue field
+	public int getBeatValue() {
+		return beatValue;
+	}
+
+	// Sets the beatValue field to the value of the input parameter
+	public void setBeatValue(int beatValue) {
+		this.beatValue = beatValue;
+	}
+	
+	// Returns rhythmicValues field
+	public Rhythmic_Value[] getRhythmicValues() {
+		return rhythmicValues;
+	}
+
+	// Sets the rhythmicValues field to the value of the input parameter
+	public void setRhythmicValues(Rhythmic_Value[] rhythmicValues) {
+		this.rhythmicValues = rhythmicValues;
+	}
+	
+	// Initialises Each Rhythmic Block's rhythmicValues array with the appropriate Rhythmic Values
+	public void initialise(){
+		
+		String name = this.name();
+		if(name == "END"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.MINIM_REST;
+		}
+		else if(name == "LASTNOTE"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.MINIM;
+		}
+		else if(name == "MINIM1"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.MINIM;
+		}
+		else if(name == "CROTCHETS1"){
+			rhythmicValues = new Rhythmic_Value [1];
+			rhythmicValues[0] = Rhythmic_Value.CROTCHET;
+		}
+		else if(name == "CROTCHETS2"){
+			rhythmicValues = new Rhythmic_Value [2];
+			rhythmicValues[0] = Rhythmic_Value.CROTCHET;
+			rhythmicValues[1] = Rhythmic_Value.CROTCHET;
+		}
+		else if(name == "CROTCHET_SWUNG_T1"){
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.CROTCHET;
+			rhythmicValues[1] = Rhythmic_Value.QUAVER;
+			rhythmicValues[2] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[3] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			
+		}
+		else if(name == "CROTCHET_SWUNG_T2"){
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.CROTCHET;
+			
+		}
+		else if(name == "CROTCHET_SWUNG_T3"){
+			rhythmicValues = new Rhythmic_Value [5];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.QUAVER;
+			rhythmicValues[4] = Rhythmic_Value.QUAVER;
+			
+		}
+		else if(name == "CROTCHET_SWUNG_T4"){
+			rhythmicValues = new Rhythmic_Value [6];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.QUAVER;
+			rhythmicValues[4] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[5] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			
+		}
+		else if(name == "TRIPLET_CROTCHETS_FULL"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.